lib/insights.ts

```diff
@@ -45,14 +52,16 @@ class AlgoliaAnalytics {
   private processQueue: () => void;
   private sendEvent: (
     eventType: InsightsEventType,
-    data: InsightsSearchClickEvent | InsightsSearchConversionEvent
+    data: InsightsEvent
```

Comment:
It's better to have more restrictive types rather than a general one like this. It's fine for the moment but it would be nice to review the architecture of the lib with type first in mind at some point.

Reply:
yes, I would love to have your input on the types.